src/client/csfle/client_encryption.rs

```diff
@@ -62,8 +64,8 @@ impl ClientEncryption {
     /// Returns the _id of the created document as a UUID (BSON binary subtype 0x04).
     pub async fn create_data_key(
         &self,
-        kms_provider: KmsProvider,
-        opts: impl Into<Option<DataKeyOptions>>,
+        kms_provider: &KmsProvider,
```

Comment:
Not strictly a bugfix, but I realized from usage that there's no reason this (and other similar methods) need to consume the args, and it's often convenient to be able to reuse options.

Reply:
I'm a little bit concerned about the inconsistency between this and our other methods that take options, since in all those other cases we accept them by value. Based on your usage thus far, is accepting a reference here (for `DataKeyOptions` at least) a big benefit? If not, then it might be better to keep it by value to prevent user compilation errors in the future. If it saves a ton of copying in a hot path or otherwise makes the users life a lot easier, then I think the inconsistency would be okay.

Reply:
It's not likely to be part of a hot path, but it is convenient when creating multiple data keys from a given provider (see the various `custom_endpoint_*` tests).  The ergonomics win isn't huge, it's just saving a `.clone()`, but the inconsistency seems like a very minor concern in this case.

Reply:
In practice, I agree it's not a huge concern. That said, at an abstract level, strict API consistency is somewhat important, even if in an intangible way. On a more practical level, my concerns would be:
- if in the future we want to add new options to `DataKeyOptions` that we want to take by value, we can't
- users will need to refer to the documentation for each method or their compiler output to know whether they need to pass options in by reference or not, which might be a poor ergonomic experience.

If performance is a motivating factor here, I think I'd personally lean towards making the options struct lighter (e.g. by `Arc`-ing constant and/or big fields) rather than taking it in by reference. If performance isn't a concern, then I lean towards strict consistency.

I think it's totally fine to just take `KmsProvider` in by reference though. `EncryptOptions` I think I lean towards taking by value also for similar reasons, though given that it is required and not an `Option`, I think consistency is less of a factor there.

All this being said, happy to leave this as-is the team doesn't really see this as being a concern.